wp/headless-wp/includes/classes/Preview/preview.php

```diff
@@ -38,5 +38,5 @@
 	$locale
 );
 
-wp_redirect( $preview_url );
+wp_safe_redirect( $preview_url );
```

Comment:
Same thing as my preview comments, we would need to add the headless site url to `allowed_redirect_hosts`.